### ISSUE TITLE
[TextInputLayout] Changing endIconDrawable ignores endIconTintList

### ISSUE THREAD

**Issue by mpost** [material-components/material-components-android#1142]
**Description:** Setting the endIconDrawable in an initial xml file or in code works as expected. Changing it afterwards does not apply to endIconTintList to the drawable.

**Expected behavior:** Always apply the the endIconTintList to the endIconDrawable when it changes.

**Source code:**
```kt
Handler().postDelayed({
  inputLayout.endIconDrawable = context.getDrawable(R.drawable.ic_icon)
}, 3000)
```

**Android API version:** Api 29

**Material Library version:** 1.2.0-alpha05

**Device:** emulator

**Comment by shekibobo** [comment]
It looks like this was re-introduced somewhere between 1.2.1 and 1.3.0-alpha03, and remains an issue in the latest 1.4.0 beta.

Since we updated to 1.3.0 stable, we are unable to show our actual startIconDrawable on a `TextInputLayout` (it always shows the tint using our primary color in the shape of the drawable instead), and we are unable to set our `startIconTintList(null)` has no apparent effect on the icon itself. From debugging, it appears the tint list gets set, but just does not get applied to the icon.

I think this was likely introduced in https://github.com/material-components/material-components-android/commit/33700b0cd35a41914dccf2f654d750209bc4c254#, while an attempted fix was tried in https://github.com/material-components/material-components-android/commit/855602bb4f733cd7a0defc6f6e1e01d13cf13f35#diff-92a36b2550[…]3a3a95856f9795f

Basically if we dynamically set a start icon on an TextInputLayout, even if we programatically set the tint of the start icon to null, it still applies the default `colorPrimary` tint to the drawable.

**Comment by hegazy** [comment]
I have this same issue with the `setStartIconDrawable` method. it sets the start icon correctly, but the tint is never applied!
API version: 30
Material lib version: 1.4.0-rc01

Device: Emulator

**Comment by davidmcraejr** [comment]
Has anyone found a way to make this work?

**Comment by leticiarossi** [comment]
the icon tints not working as expected has been fixed in https://github.com/material-components/material-components-android/commit/26d862bc5433eb50ba0875e136ff937d4e5ae617, if there are still issues please file a new bug with more info!
